src/lib/modals/Modal.svelte

```diff
@@ -114,17 +107,23 @@
 </script>
 
 {#if open}
+  <!-- backdrop -->
+  <div class={classNames('fixed inset-0 z-40', backdropClasses)} />
+  <!-- dialog -->
   <div
-    tabindex="-1"
-    class={mainClass}
-    aria-modal="true"
-    role="dialog"
     on:keydown|preventDefault={handleKeys}
     on:wheel|preventDefault
     use:prepareFocus
     use:focusTrap
-    on:click={autoclose ? onAutoClose : null}>
-    <div class="flex p-4 w-full {sizes[size]} h-full md:h-auto max-h-screen">
+    on:click={autoclose ? onAutoClose : null}
+    class={classNames(
+      'fixed top-0 left-0 right-0 h-modal md:inset-0 md:h-full z-50 w-full p-4 flex',
```

Comment:
(Continuation of the above comment)  
  
... `fixed top-0 left-0 right-0 h-modal`... are all backdrop related classes If so, shouldn't they belong to the first \`div\` which you just added?

Reply:
The implementation tries to follow the Flowbite.com layout:
```svelte
<!-- backdrop - full screen semi-transparent layer to dim the screen -->
<div class="fixed inset-0 z-40  ... colors" />

<!-- dialog -->

<!-- first, almost full screen layer used only to center (or position) the dialog:
  - in the desktop view it's full screen and seems like a duplication of the above,
  - but in the mobile view the height is 'h-modal` i.e. slightly smaller then full screen leaving small space at the bottom -->
<div class"fixed top-0 left-0 right-0 h-modal md:inset-0 md:h-full z-50 w-full p-4 flex ...getPlacementClasses()">

  <!-- second, the layer to control the dialog size -->
  <div class="relative {sizes[size]} max-h-full flex">

    <!-- end finally the Modal content with no sizing settings -->
      <Frame rounded shadow class='relative flex flex-col'>
       ....
      </Frame>
  </div>
</div>
```

Reply:
Thanks for the explanation. Now, this makes sense seeing the backdrop  `div` you added was a sibling and not a parent.

Also, the explanation of the `h-modal` now is clear that it is for the mobile view to start with.